From 1d1f9bef968cd377e1668c08e976c70b4be2483f Mon Sep 17 00:00:00 2001
From: Peter Geis <pgwipeout@gmail.com>
Date: Thu, 22 Jul 2021 20:55:31 -0400
Subject: [PATCH 096/478] arm64: dts: rockchip: add usb3 nodes to quartz64
 model a

Add the usb3 nodes to the quartz64 model a device tree.
THESE ARE NOT FUNCTIONAL YET.

Signed-off-by: Peter Geis <pgwipeout@gmail.com>
---
 .../boot/dts/rockchip/rk3566-quartz64-a.dts   | 42 +++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts b/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
index 798adafed7e5..7b1cacfe961b 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
@@ -152,6 +152,16 @@ vcc5v0_usb20_host: vcc5v0_usb20_host {
 		vin-supply = <&vcc5v0_usb>;
 	};
 
+	vcc5v0_usb20_otg: vcc5v0_usb20_otg {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc5v0_usb20_otg";
+		enable-active-high;
+		gpio = <&gpio4 RK_PB5 GPIO_ACTIVE_HIGH>;
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		vin-supply = <&dcdc_boost>;
+	};
+
 	vcc3v3_pcie_p: vcc3v3_pcie_p {
 		compatible = "regulator-fixed";
 		enable-active-high;
@@ -663,6 +673,16 @@ &uart2 {
 	status = "okay";
 };
 
+&u2phy0_host {
+	phy-supply = <&vcc5v0_usb20_host>;
+	status = "okay";
+};
+
+&u2phy0_otg {
+	phy-supply = <&vcc5v0_usb20_otg>;
+	status = "okay";
+};
+
 &u2phy1_host {
 	phy-supply = <&vcc5v0_usb20_host>;
 	status = "okay";
@@ -673,10 +693,32 @@ &u2phy1_otg {
 	status = "okay";
 };
 
+&usb2phy0 {
+	status = "okay";
+};
+
 &usb2phy1 {
 	status = "okay";
 };
 
+&usbdrd_dwc3 {
+	status = "okay";
+};
+
+&usbdrd30 {
+	status = "okay";
+};
+
+/* usb3 controller is muxed with sata1 */
+&usbhost_dwc3 {
+	status = "disabled";
+};
+
+/* usb3 controller is muxed with sata1 */
+&usbhost30 {
+	status = "disabled";
+};
+
 &usb_host0_ehci {
 	status = "okay";
 };
-- 
2.35.3

